firmware/Core/Inc/temperature_sensor/temperature_sensor.h

```diff
@@ -0,0 +1,20 @@
+#ifndef __INCLUDE_GUARD_TEMPERATURE_SENSOR_H__
+#define __INCLUDE_GUARD_TEMPERATURE_SENSOR_H__
+
+#include <stdint.h>
+
+typedef enum  
+{
+    ONE_BIT_PRECISION = 0,
+    TWO_BIT_PRECISION,
+    THREE_BIT_PRECISION,
+    FOUR_BIT_PRECISION,
+} DataPrecision;
+
+int read_temperature(float *result);
```

Comment:
I think this should return a uint8_t because the return values are always >=0

Reply:
Changed.